eog/eogtricks-quickmove.py

```diff
@@ -0,0 +1,134 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+from __future__ import print_function
+
+import re
+import os
+import shutil
+import logging
+
+from gi.repository import Eog
+from gi.repository import GObject
+from gi.repository import Gio
+from gi.repository import Gtk
+from gi.repository import Pango
+from gi.repository import GLib
+
+
+logger = logging.getLogger(__name__)
+if os.environ.get("EOGTRICKS_DEBUG"):
+    logging.basicConfig(level=logging.DEBUG)
+
+class QuickMove(GObject.GObject, Eog.WindowActivatable):
+
+    ACTION_NEW_NAME = "new-quick-move-folder"
+    ACTION_MOVE_NAME = "do-quick-move"
+
+    window = GObject.property(type=Eog.Window)
+    folder = None # os.path.expanduser('~')
+
+    def __init__(self):
+        super().__init__()
+        self.action_new = Gio.SimpleAction(name=self.ACTION_NEW_NAME)
+        self.action_move = Gio.SimpleAction(name=self.ACTION_MOVE_NAME)
+        self.action_new.connect("activate", self._new_activated_cb)
+        self.action_move.connect("activate", self._move_activated_cb)
+
+    def do_activate(self):
+        logger.debug("Activated. Adding action win.%s", self.ACTION_NEW_NAME)
+        logger.debug("Activated. Adding action win.%s", self.ACTION_MOVE_NAME)
+        self.window.add_action(self.action_new)
+        self.window.add_action(self.action_move)
+        app = self.window.get_application()
+        app.set_accels_for_action( "win." + self.ACTION_NEW_NAME, ['N'], )
+        app.set_accels_for_action( "win." + self.ACTION_MOVE_NAME, ['M'], )
+        self.window.get_titlebar().set_subtitle("Target: None")
+
+    def do_deactivate(self):
+        logger.debug("Deactivated. Removing action win.%s", self.ACTION_NEW_NAME)
+        logger.debug("Deactivated. Removing action win.%s", self.ACTION_MOVE_NAME)
+        self.window.remove_action(self.ACTION_NEW_NAME)
+        self.window.remove_action(self.ACTION_MOVE_NAME)
+
+    def _move_activated_cb(self, action, param):
+        if not self.folder:
+            return
+
+        img = self.window.get_image()
+        if not img:
+            return
+        if not img.is_file_writable():
+            return
+
+        src = img.get_file().get_path()
+        name = os.path.basename(src)
+        dest = self.folder
+
+        # Create directory if it doesn't exist.
+        try:
+            os.makedirs(dest)
+        except OSError:
+            pass
+
+        # If you rename the current image, the image is
+        # re-inserted at its new aphabetical location, and the
+        # UI's idea of the current image resets to position
+        # zero. This is confusing and makes things feel really
+        # inconsistent.
```

Comment:
Into the same folder? I can see why it might do that.

Perhaps check for `src == dest` (after some `os.path` cleanup) and take no view manipulation actions or move actions if they're the same?

Reply:
Thanks, fixed that. Also found a corner case when you cancel the folder selection dialog.